.circleci/config.yml

```
@@ -308,6 +308,51 @@ jobs:
               "${ARNOLD_IMAGE}" \
               python -m pytest --no-cov
 
+  # ---- DockerHub publication job ----
+  hub:
+    # We use the machine executor, i.e. a VM, not a container
+    machine:
+      # Cache docker layers so that we strongly speed up this job execution
+      docker_layer_caching: true
+
+      working_directory: ~/fun
+
+    steps:
+      - checkout
+      - *attach_workspace
+      - *docker_load
+      - *ci_env
+      # Login to DockerHub to Publish new images
+      #
+      # Nota bene: you'll need to define the following secrets environment vars
+      # in CircleCI interface:
+      #
+      #   - DOCKER_USER
+      #   - DOCKER_PASS
+      - run:
+          name: Login to DockerHub
+          command: echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
+
+      # Tag docker images with the same pattern used in Git (Semantic Versioning)
+      #
+      # Git tag: v1.0.1
+      # Docker tag: 1.0.1
+      - run:
+          name: Tag images
+          command: |
+            DOCKER_TAG=$(echo ${CIRCLE_TAG} | sed 's/^v//')
```

To ease CI debugging, I often add an image list:

```suggestion
            docker images fundocker/arnold
            DOCKER_TAG=$(echo ${CIRCLE_TAG} | sed 's/^v//')
```

---

:+1: